### PR TITLE
Implement button component

### LINKED ISSUE #5: Implement the button component
### Adobe XD
https://xd.adobe.com/view/fdd0941a-c327-46ee-b303-1c4618143fbc-a883/

### PR DESCRIPTION
**Pull request by 0xfr4nkie**
This pull request includes the implementation of the button component according to the specs from Adobe XD. 

In addition, this PR also includes an addition to the preview.js file which defines the background colors for storybook. The color theme matches the ASU theme.

Lastly, settings.json within .vscode has been updated to include scss to use prettier as its default formatter.

### CODE REVIEW

File: packages/ui-styles/scss/lib/_colors.scss [hunk 1]
```diff
@@ -32,3 +32,8 @@ $color-secondary-inverted: $color-secondary;
 $color-background: $color-white;
 $color-background-light: $color-background;
 $color-background-inverted: $color-black;
+
+// ASU colors
```
Comment (KevinThePepper) [inline packages/ui-styles/scss/lib/_colors.scss:36]: More just an FYI. No action required... If we really want to match ASU colors, we can use [their color palette](https://brandguide.asu.edu/brand-elements/design/color).

File: packages/ui/src/components/elements/Button/Button.tsx [hunk 2]
```diff
@@ -0,0 +1,32 @@
+import React from "react";
+import cx from "clsx";
+
+import ButtonProps from "./Button.props";
+import { UI_CLASSNAME } from "../../..";
+
+function Button({
```
Comment (KevinThePepper) [inline packages/ui/src/components/elements/Button/Button.tsx]: @frankjlin16 can you change this from function to arrow notation? See [this example](https://asu-codedevils.github.io/CD-Library/code-standards#react-component)

File: packages/ui/src/components/elements/Button/Button.tsx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  className,
+  variant,
+  inverted,
+  disabled,
+  label,
+}: ButtonProps) {
+  //TODO: Add your code here
```
Comment (KevinThePepper) [inline packages/ui/src/components/elements/Button/Button.tsx]: @frankjlin16 can you remove this comment?

File: packages/ui/src/components/elements/Button/Button.tsx [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+import ButtonProps from "./Button.props";
+import { UI_CLASSNAME } from "../../..";
+
+function Button({
+  className,
+  variant,
+  inverted,
+  disabled,
+  label,
+}: ButtonProps) {
```
Comment (KevinThePepper) [inline packages/ui/src/components/elements/Button/Button.tsx]: @frankjlin16 can you add a `...htmlButtonProps` attribute to both the props and the button? See [this example](https://asu-codedevils.github.io/CD-Library/code-standards#react-component). These are the built-in HTML `button` props and we want people to be able to access these.

File: packages/ui/src/components/elements/Button/Button.tsx [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+  //TODO: Add your code here
+  return (
+    <div>
```
Comment (KevinThePepper) [inline packages/ui/src/components/elements/Button/Button.tsx]: @frankjlin16 is there a use case for this outer `div`? If not can it be removed?

File: packages/ui-styles/scss/lib/_mixins.scss [hunk 6]
```diff
@@ -5,3 +5,9 @@
   -webkit-border-radius: $radius;
   -moz-border-radius: $radius;
 }
+
+@mixin button-border-radius($radius: $button-border-radius) {
```
Comment (KevinThePepper) [inline packages/ui-styles/scss/lib/_mixins.scss]: @frankjlin16 great idea! Can you make this more generic and just call it `border-radius`?

File: packages/ui-styles/scss/components/button/_button.scss [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+    font: $font-button;
+
+    &.primary {
+      background-color: $color-asu-maroon;
+      color: $color-white;
+      border: none;
+      padding: 10px 20px;
+
+      &:hover {
+        background-color: $color-better-black;
```
Comment (KevinThePepper) [inline packages/ui-styles/scss/components/button/_button.scss:28]: @frankjlin16 can we add a transition here and in `.secondary.hover`?

```css
transition: all 0.3s ease-out;
```

**Review by KevinThePepper — CHANGES_REQUESTED** [review]

File: packages/ui-styles/scss/lib/_colors.scss (hunk 1 above)
Comment (jeromeraphael) [reply packages/ui-styles/scss/lib/_colors.scss:36]: Can we reuse the values that's on the XD file

Comment (0xfr4nkie) [reply packages/ui-styles/scss/lib/_colors.scss:36]: > More just an FYI. No action required... If we really want to match ASU colors, we can use [their color palette](https://brandguide.asu.edu/brand-elements/design/color).

Ooo, nice. I will keep that in mind for future components! Thanks!

**Review by KevinThePepper — APPROVED** [review]

**Pull request merged**